### PR TITLE
- Remove maven-scr plugin

### LINKED ISSUE #84: Get rid of maven-scr-plugin and only use OSGi R6 annotations
Since bnd supports those natively the maven-bundle-plugin (in a recent version) can be used to generate component descriptors and metatype resources.

### PR DESCRIPTION
**Pull request by bpauli**
- Adding OSGi annotations
- Adapt necessary changes to examples
- Update to latest released uber-jar
This fixes #64 and resolve #84

### CODE REVIEW

File: src/main/archetype/pom.xml [hunk 1]
```diff
@@ -482,33 +452,33 @@
         <dependencies>
             <!-- OSGi Dependencies -->
             <dependency>
-                <groupId>org.apache.felix</groupId>
-                <artifactId>org.apache.felix.scr</artifactId>
-                <version>1.6.0</version>
+                <groupId>biz.aQute</groupId>
+                <artifactId>bndlib</artifactId>
+                <version>1.50.0</version>
```
Comment (kwin) [inline src/main/archetype/pom.xml]: this shouldn't be necessary to add as dependency

File: src/main/archetype/pom.xml [hunk 2]
```diff
@@ -482,33 +452,33 @@
...
                 <scope>provided</scope>
             </dependency>
             <dependency>
-                <groupId>org.apache.felix</groupId>
-                <artifactId>org.apache.felix.scr.annotations</artifactId>
-                <version>1.9.6</version>
+                <groupId>org.osgi</groupId>
+                <artifactId>org.osgi.core</artifactId>
+                <version>4.2.0</version>
```
Comment (kwin) [inline src/main/archetype/pom.xml]: I think it makes more sense to target OSGi Core/Compendium 6.0, as we leverage metatype and component annotations which are only compatible with OSGi Core/Compendium 6.0.

File: src/main/archetype/pom.xml [hunk 3]
```diff
@@ -482,33 +452,33 @@
...
                 <scope>provided</scope>
             </dependency>
             <dependency>
-        <groupId>biz.aQute</groupId>
-        <artifactId>bndlib</artifactId>
-        <version>1.50.0</version>
-        <scope>provided</scope>
+                <groupId>org.osgi</groupId>
```
Comment (kwin) [inline src/main/archetype/pom.xml]: I think it makes more sense to target OSGi Core/Compendium 6.0, as we leverage metatype and component annotations which are only compatible with OSGi Core/Compendium 6.0

File: src/main/archetype/core/pom.xml [hunk 4]
```diff
@@ -76,6 +73,14 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.service.component.annotations</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.service.metatype.annotations</artifactId>
+        </dependency>
         <!-- Other Dependencies -->
```
Comment (kwin) [inline src/main/archetype/core/pom.xml]: Explicitly adding the versioning annotations from https://mvnrepository.com/artifact/org.osgi/org.osgi.annotation/6.0.0 makes sense as well. The Bnd ones are deprecated now.

**Comment by rombert** [comment]
@bpauli  - would you be able to update the PR based on @kwin's comments? I think it makes sense to use the latest versions of the annotations and the OSGi libraries since we now explicitly target 6.2 with this version.

**Comment by bpauli** [comment]
@kwin Thank you for your review.

@rombert I updated the PR accordingly

File: src/main/archetype/core/pom.xml [hunk 5]
```diff
@@ -57,24 +62,24 @@
     <dependencies>
         <!-- OSGi Dependencies -->
         <dependency>
-            <groupId>org.apache.felix</groupId>
-            <artifactId>org.apache.felix.scr</artifactId>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.core</artifactId>
```
Comment (kwin) [inline src/main/archetype/core/pom.xml]: I am pretty sure the right artifactId is `osgi.core` (http://search.maven.org/#artifactdetails%7Corg.osgi%7Cosgi.core%7C6.0.0%7Cjar) and not `org.osgi.core`. Compare also with http://blog.osgi.org/2015/08/release-6-of-osgi-compendium-osgi.html.

File: src/main/archetype/core/pom.xml [hunk 6]
```diff
@@ -57,24 +62,24 @@
...
-            <groupId>org.apache.felix</groupId>
-            <artifactId>org.apache.felix.scr.annotations</artifactId>
+            <groupId>org.osgi</groupId>
+            <artifactId>osgi.cmpn</artifactId>
         </dependency>
         <dependency>
-            <groupId>biz.aQute</groupId>
-            <artifactId>bndlib</artifactId>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.service.component.annotations</artifactId>
```
Comment (kwin) [inline src/main/archetype/core/pom.xml]: the annotations are part of  `osgi.cmp` and don't need to be added as separate dependency. There shouldn't be the same class twice in the classpath.

File: src/main/archetype/core/pom.xml [hunk 7]
```diff
@@ -57,24 +62,24 @@
...
         </dependency>
         <dependency>
             <groupId>org.osgi</groupId>
-            <artifactId>org.osgi.core</artifactId>
+            <artifactId>org.osgi.service.metatype.annotations</artifactId>
```
Comment (kwin) [inline src/main/archetype/core/pom.xml]: the annotations are part of  `osgi.cmp` and don't need to be added as separate dependency. There shouldn't be the same class twice in the classpath.

**Comment by bpauli** [comment]
@kwin Thanks for hint. Fixed with my latest commit.

**Comment by rombert** [comment]
Manually merged in 799b3fa30273c18e5757e1a008f44cdb15e3d7cb, thanks! Would you mind taking a stab at removing the scr-maven-plugin from the `it.tests` module as well?

**Pull request closed without merging**